src/components/SiteHeader.js

```diff
@@ -156,11 +195,20 @@ const SiteHeader = () => {
                                 {openIdConfigFetching || isHandingOffCodeForToken ? "Loading..." : "Sign In"}
                             </span>
                         ),
-                        onClick: () => performAuth(authzEndpoint, CLIENT_ID, AUTH_CALLBACK_URL),
+                        onClick: () => performAuth(),
                     },
                 ]),
         ],
-        [isAuthenticated, authHasAttempted, idTokenContents, performSignOut, unreadNotifications],
+        [
+            isAuthenticated,
+            authHasAttempted,
+            idTokenContents,
+            performSignOut,
+            performAuth,
+            canQueryData,
+            managerPermissions,
+            unreadNotifications,
```

Comment:
Missing: `openIdConfigFetching`, `isHandingOffCodeForToken`, `isFetchingPermissions`, `canViewNotifications`, `hasAttemptedQueryPermissions`, `hasAttemptedManagerPermissions`.

We should probably add the eslint rules for react hooks and dependency array, this way the linter can do these checks for us.
"react-hooks/rules-of-hooks": "error", // For checking rules of hooks
"react-hooks/exhaustive-deps": "warn" // For checking hook dependencies

Reply:
good idea - i'll make a separate PR after this for that I think